lib/ret_web/controllers/upload_controller.ex

```diff
@@ -9,30 +17,33 @@ defmodule RetWeb.UploadController do
     handle(conn, upload_id)
   end
 
-  defp handle(conn, upload_id) do
-    case conn |> get_req_header("authorization") do
-      [<<"Token ", token::binary>>] ->
-        case Ret.Uploads.fetch(upload_id, token) do
-          {:ok, %{"content_type" => content_type, "content_length" => content_length}, stream} ->
-            conn =
-              conn
-              |> put_resp_content_type(content_type)
-              |> put_resp_header("content-length", "#{content_length}")
-              |> put_resp_header("transfer-encoding", "chunked")
-              |> put_resp_header("cache-control", "public, max-age=31536000")
-              |> send_chunked(200)
+  defp fetch_upload(conn, upload_id, token) do
+    case Ret.Uploads.fetch(upload_id, token) do
+      {:ok, %{"content_type" => content_type, "content_length" => content_length}, stream} ->
+        conn =
+          conn
+          |> put_resp_content_type(content_type)
+          |> put_resp_header("content-length", "#{content_length}")
+          |> put_resp_header("transfer-encoding", "chunked")
+          |> put_resp_header("cache-control", "public, max-age=31536000")
+          |> send_chunked(200)
 
-            stream |> Stream.map(&chunk(conn, &1)) |> Stream.run()
+        stream |> Stream.map(&chunk(conn, &1)) |> Stream.run()
 
-            conn
+        conn
 
-          {:error, :not_found} ->
-            conn |> send_resp(400, "")
+      {:error, :not_found} ->
+        conn |> send_resp(400, "")
 
-          {:error, :not_allowed} ->
-            conn |> send_resp(401, "")
-        end
+      {:error, :not_allowed} ->
+        conn |> send_resp(401, "")
+    end
+  end
 
+  defp handle(conn, upload_id) do
```

Comment:
I'd either rename this to `extract_header_and_fetch_upload` or just make it `fetch_upload/2`